### PR TITLE
TW-578/NFT-header

### PR DESCRIPTION
**Pull request by Boltman92**

### CODE REVIEW

**Review by svjatoslavtt — DISMISSED** [review]

File: src/screens/settings/settings-header/social-button/social-button.tsx [hunk 1]
```diff
@@ -11,14 +12,17 @@ import { useSocialButtonStyles } from './social-button.styles';
...
 }
 
-export const SocialButton: FC<Props> = ({ iconName, url }) => {
+export const SocialButton: FC<Props> = ({ iconName, url, style, color, size = formatSize(24) }) => {
   const styles = useSocialButtonStyles();
 
   return (
-    <TouchableOpacity style={styles.container} onPress={() => openUrl(url)}>
-      <Icon name={iconName} size={formatSize(24)} />
+    <TouchableOpacity style={[styles.container, style]} onPress={url ? () => openUrl(url) : undefined}>
```
Comment (svjatoslavtt) [inline src/screens/settings/settings-header/social-button/social-button.tsx:24]: can we make a separate function?

File: src/screens/collectibles-home/collectibles-home.tsx [hunk 2]
```diff
@@ -1,47 +1,130 @@
...
+          <CurrentAccountDropdown
+            value={selectedAccount}
+            list={visibleAccounts}
+            onValueChange={onValueChange}
+            isCollectibleScreen
+          />
+        </View>
+        <View style={styles.profileContainer}>
+          <View style={styles.createProfile}>
+            <TouchableIcon name={IconNameEnum.PlusCircle} onPress={() => null} size={formatSize(16)} />
```
Comment (svjatoslavtt) [inline src/screens/collectibles-home/collectibles-home.tsx:75]: Maybe we need to add common EmptyFn

File: src/screens/collectibles-home/collectibles-home.tsx [hunk 3]
```diff
@@ -1,47 +1,130 @@
...
   usePageAnalytic(ScreensEnum.CollectiblesHome);
 
-  const collectiblesList = useVisibleCollectiblesListSelector();
+  useEffect(() => void dispatch(loadCollectionsActions.submit(selectedAccount.publicKeyHash)), [selectedAccount]);
+
+  const onValueChange = (value: AccountBaseInterface | undefined) =>
+    dispatch(setSelectedAccountAction(value?.publicKeyHash));
+
+  const renderItem: ListRenderItem<Collection> = ({ item }) => (
+    <TouchableOpacity style={styles.collectionBlock} onPress={() => openUrl(OBJKT_COLLECTION_URL(item.contract))}>
```
Comment (svjatoslavtt) [inline src/screens/collectibles-home/collectibles-home.tsx:47]: can we make a separate function, too

File: src/screens/collectibles-home/collectibles-home.tsx [hunk 4]
```diff
@@ -1,47 +1,130 @@
...
+  const collections = useCreatedCollectionsSelector();
+
+  const selectedAccount = useSelectedAccountSelector();
+  const visibleAccounts = useVisibleAccountsListSelector();
+  const colors = useColors();
 
   usePageAnalytic(ScreensEnum.CollectiblesHome);
 
-  const collectiblesList = useVisibleCollectiblesListSelector();
+  useEffect(() => void dispatch(loadCollectionsActions.submit(selectedAccount.publicKeyHash)), [selectedAccount]);
```
Comment (svjatoslavtt) [inline src/screens/collectibles-home/collectibles-home.tsx:41]: I think we need to change dependencies to `[selectedAccount.publicKeyHash]`

File: src/store/collectons/collections-reducers.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { createReducer } from '@reduxjs/toolkit';
+
+import { loadCollectionsActions } from './collections-actions';
+import { Collection, collectionsInitialState, CollectionState } from './collections-state';
+
+export const collectionsReducer = createReducer<CollectionState>(collectionsInitialState, builder => {
+  builder.addCase(loadCollectionsActions.success, (state, { payload: newCollections }) => {
+    const result = newCollections.reduce<Record<string, Collection[]>>((acc, current) => {
+      if (acc[current.creator] !== undefined) {
```
Comment (svjatoslavtt) [inline src/store/collectons/collections-reducers.ts:9]: not necessary: I think it looks a little better `current.creator in acc` :)

**Review by svjatoslavtt — DISMISSED** [review]

**Review by svjatoslavtt — APPROVED** [review]

**Review by keshan3262 — DISMISSED** [review]

**Review by keshan3262 — DISMISSED** [review]

**Review by keshan3262 — DISMISSED** [review]

**Review by keshan3262 — DISMISSED** [review]

**Review by keshan3262 — APPROVED** [review]

**Review by keshan3262 — DISMISSED** [review]

**Pull request merged**